src/controller/Controller.ts

```diff
@@ -0,0 +1,74 @@
+import View from '../view/View';
+import Computer from '../model/Computer';
+import { GAME_CODE } from '../constants/game';
+import { MESSAGE } from '../constants/message';
+import { convertToNumberArray } from '../utils/convert';
+import {
+  hasRepeatedNumbers,
+  isValidCode,
+  isValidLength,
+  isValidNumbers,
+} from '../utils/validation';
+
+class Controller {
+  #computer;
+
+  constructor() {
+    this.#computer = new Computer();
+  }
+
+  start = () => {
+    View.printStartMessage();
+    this.compare();
+  };
+
+  compare = () => {
+    View.inputPlayerNumbers((input: string) => {
+      this.inputValidation(input);
+
+      const playerNumbers = convertToNumberArray(input);
+
+      const count = this.#computer.compareNumbers(playerNumbers);
+      const hintMessage = this.#computer.hintMessage(count);
+
+      View.printMessage(hintMessage);
+
+      const isWin = count.strike === GAME_CODE.WIN;
+      if (isWin) {
+        View.printMessage(MESSAGE.WIN_GAME);
+        this.askRestart();
+      } else {
+        this.compare();
+      }
```

Comment:
개인적으로 궁금한 부분인데, 저는 함수를 짤 때 선언부 - 사용부를 나누어서 생각을 하여 안에 로직을 분리합니다.
그래서 제가 이 파트를 짰다면 다음과 같이 짰을 것 같아요! 다른 의견 주시면 감사하겠습니다👍

```suggestion
      this.inputValidation(input);

      const playerNumbers = convertToNumberArray(input);
      const count = this.#computer.compareNumbers(playerNumbers);
      const hintMessage = this.#computer.hintMessage(count);

      View.printMessage(hintMessage);

      if (count.strike === GAME_CODE.win) {
        View.printMessage(MESSAGE.win_game);
        this.askRestart();
      } else {
        this.compare();
      }
```
이런식으로 작성했을 것 같아요. isWin같은 경우는 isWin이 있으면 한눈에 알아보기 쉽겠지만,
이미 조건식 자체에 GAME_CODE.win 과 비교를 하는 부분에서 없어도 빠르게 유추되지 않을까 싶어서 제거했습니다!

아 그리고 위에다가 안적어놓은게 있었는데 보통 객체로 상수들을 관리하는 경우에는 일반적으로 외부에는 대문자를 사용하지만, 내부에는 소문자로 작성하더라구요 (google js convention에서는 반대로 외부에 소문자 내부에 대문자를 사용해요)

관련 링크 첨부해드립니다! [Naming convention for const object keys in es6](https://stackoverflow.com/questions/40291766/naming-convention-for-const-object-keys-in-es6)

Reply:
코드 작성 시 선언부-사용부를 나눠서 적용한다는 의견 감사합니다.
코드 분리할 때 항상 고민하는 부분인데, 명확한 구분을 할 수 있는 의견이라 적용해보기 좋을 것 같아요!
저는 기능에 따라 분리를 해보려고 하는데, 기능이 연관되어 있으면 분리할 때 굉장히 애매해서 일관성있게 분리하지 못했네요.

조건문도 자주 고민하는 부분인데 의견 감사합니다.
현호님의 예시가 이해하기 쉽다고 생각이 들면서도 사람에 따라 다를 수 있지 않을까 생각하게 되는 것 같아요!

상수 관련 네이밍에 대해 참고를 못찾았었는데 감사합니다. 잘 읽어볼게요!